autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+                        "before feeding it to Auto-Sklearn.".format(
+                            column,
+                            X[column].dtype.name,
+                        )
+                    )
+            else:
+                feature_types.append('numerical')
+
+        # Make sure we only set this once. It should not change
+        if not self.feature_types:
+            self.feature_types = feature_types
+
+        # This proc has to handle multiple calls, for X_train
+        # and X_test scenarios. We have to make sure also that
+        # data is consistent within calls
+        if enc_columns:
+            if self.enc_columns and self.enc_columns != enc_columns:
+                raise ValueError(
+                    "Validation was already performed on input with enc columns={} "
+                    "yet, another set of features with columns to encode={} was "
+                    "provided. Feature changing after fit is not allowed.".format(
+                        enc_columns,
+                        self.enc_columns
+                    )
+                )
+            else:
+                self.enc_columns = enc_columns
+
+            if not self.feature_encoder:
+                if self.feature_encoder_type == 'OneHotEncoder':
+                    encoder = preprocessing.OneHotEncoder(sparse=False)
+                elif self.feature_encoder_type == 'OrdinalEncoder':
+                    encoder = preprocessing.OrdinalEncoder()
+                else:
+                    raise ValueError(
+                        "Unsupported feature encoder provided {}".format(
+                            self.feature_encoder_type
+                        )
+                    )
+
+                self.feature_encoder = make_column_transformer(
+                    (encoder, self.enc_columns),
+                    remainder="passthrough"
+                )
+
+                self.feature_encoder.fit(X)
+
+        if self.feature_encoder:
+            X = self.feature_encoder.transform(X)
+
+        return X
+
+    def _validate_dataframe_target(
+        self,
+        y: Union[pd.DataFrame, np.ndarray],
+    ) -> Union[pd.DataFrame, np.ndarray]:
+        """
+        This method encodes
+        categorical series to a numerical equivalent.
+
+        No change to numerical values.
+
+        Dataframes with more than 1 column are ambiguous because
+        two column could be a label or multilabel. Force the user
+        to use numerical encoding via sklearn checks
+        """
+
+        # Convert pd.Series to dataframe as categorical series
+        # lack many useful methods
+        if isinstance(y, pd.Series):
+            y = y.to_frame().reset_index(drop=True)
+
+        # handle categorical encoding of a Dataframes
+        # with a single column
+        # A dataframe target with multiple columns is ambiguous
+        # We don't know if all columns should share same encoding
```

Comment:
I can't think of a case where they should share the same encoding, so I don't think you need to worry about this case.